File: backend/app/schemas/paper_trading.py
import json
from datetime import datetime
from pydantic import BaseModel, ConfigDict, field_validator


class PaperTradingDailyRecordBase(BaseModel):
    record_date: str
    daily_return: float
    cumulative_return: float
    nav: float


class PaperTradingDailyRecordRead(PaperTradingDailyRecordBase):
    id: int
    user_id: int
    portfolio_id: int | None
    report_id: int | None
    asset_snapshot: str | None
    created_at: datetime
    updated_at: datetime | None

    model_config = ConfigDict(from_attributes=True)


class PaperTradingMonthlyStatBase(BaseModel):
    year_month: str
    monthly_return: float
    cumulative_return_at_month_end: float
    record_count: int


class PaperTradingMonthlyStatRead(PaperTradingMonthlyStatBase):
    id: int
    user_id: int
    portfolio_id: int | None
    created_at: datetime
    updated_at: datetime | None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/paperTrading.ts
import { delay } from './mock/utils'
import { DEMO_PAPER_TRADING_DAILY_RECORDS, DEMO_PAPER_TRADING_MONTHLY_STATS } from './mock/demoData'

export interface PaperTradingDailyRecord {
  id: number
  user_id: number
  portfolio_id: number | null
  record_date: string
  daily_return: number
  cumulative_return: number
  nav: number
  report_id: number | null
  asset_snapshot: string | null
  created_at: string
  updated_at: string | null
}

export interface PaperTradingMonthlyStat {
  id: number
  user_id: number
  portfolio_id: number | null
  year_month: string
  monthly_return: number
  cumulative_return_at_month_end: number
  record_count: number
  created_at: string
  updated_at: string | null
}

export const paperTradingApi = {
  listDailyRecords(_params?: {
    portfolio_id?: number
    start_date?: string
    end_date?: string
    limit?: number
  }): Promise<PaperTradingDailyRecord[]> {
    return delay(300).then(() => DEMO_PAPER_TRADING_DAILY_RECORDS as PaperTradingDailyRecord[])
  },

  listMonthlyStats(_params?: { portfolio_id?: number; limit?: number }): Promise<PaperTradingMonthlyStat[]> {
    return delay(300).then(() => DEMO_PAPER_TRADING_MONTHLY_STATS as PaperTradingMonthlyStat[])
  },

  getLatest(_params?: { portfolio_id?: number }): Promise<PaperTradingDailyRecord | null> {
    return delay(300).then(() => {
      const records = DEMO_PAPER_TRADING_DAILY_RECORDS as PaperTradingDailyRecord[]
      return records[records.length - 1] || null
    })
  },

  syncDaily(_report_date?: string): Promise<{ detail: string; result: Record<string, unknown> }> {
    return delay(500).then(() => ({ detail: '同步完成', result: {} }))
  },

  calcMonthly(_year_month?: string): Promise<{ detail: string; result: Record<string, unknown> }> {
    return delay(500).then(() => ({ detail: '计算完成', result: {} }))
  },
}
